dts: stm32u5: move some nodes from common u5.dtsi to specific soc dtsi's.

harristomy opened the pull request:
- Moves the octospi2 and ucpd1 node into dtsi's that actually has the peripheral.
- Adds the dtsi for the stm32u5f9 soc(g9 is the AES-HW capable variant
- Corrects the memory package variants for the STM32U53(4)5xx SoCs

Related to #89704

erwango reviewed (changes requested):
Thanks for taking a stab at this, but please pay attention that some of these files are including others and changes introduce node duplication.

harristomy commented:
Ah my bad, didn't see the included headers, will refactor.

For the duplication of nodes, will follow the approach taken in the g0 series. however, the 575 and 585 have a full-speed usb peripheral , while the ones above have a high-speed. Should the fs speed node be defined in a separate dtsi, or should there be a delete-node for it in the 595.dtsi?

harristomy commented:
Same question for the octospi2 and HSPI peripheral, either some level of duplication or delete-node or separate file to include just these nodes will be required, which would you prefer?

harristomy commented:
I'll push up a version that uses the separate file method tomorrow and then revisit this.

harristomy commented:
for u595.dtsi, should there be a 595Xi.dtsi that includes `st/u5/stm32u5_usbotg_hs.dtsi>`? Not sure why the 595 breaks the pattern observed by the other micros (or if it's even correctly defined here 🤔)

erwango reviewed (dismissed):
Thanks for the updated version, but I have further comments.
One line I'd like to stick to is not to declare nodes at "memory package level".

One idea which is not perfect but pay help:

SoC .dtsi files could follow the below scheme:
------------------------------------------------------------------------ + cypto
stm32u5 -> stm32u5_fs -> stm32u535 -> stm32u575
-----------------------------------------------------> stm32u545 -> stm32u585
-------------> stm32u5_hs -> stm32u595 --------------------> stm32u5a5
-----------------------------------------------------> stm32u599 -> stm32u5a9
-----------------------------------------------------> stm32u5f9 -> stm32u5g9
----------------------------------> stm32u5f7 ------------------> stm32u5g7    
(Sorry for the `-`, but I had to add them to keep the formatting)

Then "memory package level" files (Xz.dtsi) include above files, but never add device nodes.

Note: I added SoC not yet supported to show the complete view, but no need to add them.

harristomy commented:
> Thanks for the updated version, but I have further comments. One line I'd like to stick to is not to declare nodes at "memory package level".
> 
> One idea which is not perfect but pay help:
> 
> SoC .dtsi files could follow the below scheme: ------------------------------------------------------------------------ + cypto stm32u5 -> stm32u5_fs -> stm32u535 -> stm32u575 -----------------------------------------------------> stm32u545 -> stm32u585 -------------> stm32u5_hs -> stm32u595 --------------------> stm32u5a5 -----------------------------------------------------> stm32u599 -> stm32u5a9 -----------------------------------------------------> stm32u5f9 -> stm32u5g9 ----------------------------------> stm32u5f7 ------------------> stm32u5g7 (Sorry for the `-`, but I had to add them to keep the formatting)
> 
> Then "memory package level" files (Xz.dtsi) include above files, but never add device nodes.
> 
> Note: I added SoC not yet supported to show the complete view, but no need to add them.

Sorry, I'm not sure I understand the intended structure, could you use something like https://tree.nathanfriend.com/ ?

etienne-lms commented:
IIUC, excluding the crypto part, it should look like this:
```
stm32u5
├─ stm32u5_fs
│  └─ stm32u535
│     ├─ stm32u575
│     └─ stm32u545
│        └─ stm32u585
└─ stm32u5_hs
   ├─ stm32u595
   │  ├─ stm32u5a5
   │  ├─ stm32u599
   │  │  └─ stm32u5a9
   │  └─ stm32u5f9
   │     └─ stm32u5g9
   └─ stm32u5f7
      └─ stm32u5g7

harristomy commented:
> IIUC, excluding the crypto part, it should look like this:
> 
> ```
> stm32u5
> ├─ stm32u5_fs
> │  └─ stm32u535
> │     ├─ stm32u575
> │     └─ stm32u545
> │        └─ stm32u585
> └─ stm32u5_hs
> ```
> ...

Ah I see, yep will follow that approach

harristomy committed:
dts: stm32u5: add memory package variants

Corrects stm32u53/45xx variants and adds u575Xg and u599Xi

Signed-off-by: Harris Tomy <harristomy@gmail.com>

harristomy commented:
@erwango @etienne-lms does this map to what you guys had in mind? I kept the `ltdc` node inheritance for now seeing as the g9 nucleo board uses it, and kept the `fs` and `otg_fs` inheritance separate for the 535/575 respectively, but I believe the rest should hopefully match up to your specified description.

etienne-lms commented on dts/arm/st/u5/stm32u5_crypt.dtsi:
```diff
@@ -0,0 +1,19 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Could you remove the extra empty line?

etienne-lms reviewed (commented):
LGTM. Some nitpicking comments.

harristomy commented on dts/arm/st/u5/stm32u545.dtsi:
```diff
@@ -4,33 +4,12 @@
 
-#include <st/u5/stm32u5.dtsi>
+#include <st/u5/stm32u535.dtsi>
+#include <st/u5/stm32u5_crypt.dtsi>
```
Done

harristomy commented on dts/arm/st/u5/stm32u5_usb_fs.dtsi:
```diff
@@ -0,0 +1,33 @@
+		compatible = "usb-nop-xceiv";
+		#phy-cells = <0>;
+	};
+
```
Done

harristomy commented on dts/arm/st/u5/stm32u599.dtsi:
```diff
@@ -5,9 +5,22 @@
 #include <st/u5/stm32u595.dtsi>
+#include <zephyr/dt-bindings/display/panel.h>
+#include <zephyr/dt-bindings/memory-attr/memory-attr.h>
+#include <zephyr/dt-bindings/memory-attr/memory-attr-arm.h>
```
Removed

erwango commented on dts/arm/st/u5/stm32u5f9.dtsi:
```diff
@@ -0,0 +1,23 @@
+
+/ {
+	soc {
+		compatible = "st,stm32u5f9", "st,stm32u5", "simple-bus";
```
If you're adding a dts SoC definition, please also add the matching Kconfig definition.

etienne-lms reviewed (commented):
LGTM with some nitpicking comments.

harristomy commented on dts/arm/st/u5/stm32u575.dtsi:
```diff
@@ -1,33 +1,52 @@
+#include <st/u5/stm32u5_usbotg_fs.dtsi>
+
+#include <zephyr/dt-bindings/memory-controller/stm32-fmc-nor-psram.h>
 
```
done

etienne-lms replied:
ping

harristomy replied:
Sorry, think I accidentally overrode this with one of the force pushes, should be okay now.

mathieuchopstm commented on dts/arm/st/u5/stm32u595.dtsi:
```diff
@@ -1,17 +1,52 @@
+		fmc: memory-controller@420d0400 {
+			compatible = "st,stm32-fmc";
+			reg = <0x420d0400 0x400>;
+			clocks = <&rcc STM32_CLOCK(AHB2_2, 0U)>;
```
The question should rather be "why is the `U` suffix present everywhere in this file?", to which the answer is "[because we allowed it in even though we really shouldn't have](https://github.com/zephyrproject-rtos/zephyr/pull/79683#pullrequestreview-2366609352)".
All integers in DT are unsigned - the `U` suffix is a C construct, accepted but silently discarded by `edtlib` or `dtc` (I assume this is done to allow sharing definitions between C and DTS).

harristomy replied:
Ah I see, thank you!

mathieuchopstm commented on soc/st/stm32/stm32u5x/Kconfig.defconfig.stm32u5f9xx:
```diff
@@ -0,0 +1,11 @@
+# STMicroelectronics STM32U5G9XX MCU
```
```suggestion
# STMicroelectronics STM32U5F9XX MCU
```

mathieuchopstm commented on dts/arm/st/u5/stm32u5_usb_fs.dtsi:
```diff
@@ -0,0 +1,32 @@
+
+#include <st/u5/stm32u5.dtsi>
+
+#include <zephyr/dt-bindings/clock/stm32u5_clock.h>
```
`stm32u5.dtsi` already includes it, this should not be necessary?

mathieuchopstm commented on dts/arm/st/u5/stm32u5_usbotg_hs.dtsi:
```diff
@@ -0,0 +1,33 @@
+
+#include <st/u5/stm32u5.dtsi>
+
+#include <zephyr/dt-bindings/clock/stm32u5_clock.h>
```
This should already be included by `stm32u5.dtsi`?

harristomy replied:
Removed.

mathieuchopstm commented on dts/arm/st/u5/stm32u599.dtsi:
```diff
@@ -9,5 +9,15 @@
 / {
 	soc {
 		compatible = "st,stm32u599", "st,stm32u5", "simple-bus";
+
+		ltdc: display-controller@40016800 {
+			compatible = "st,stm32-ltdc";
+			reg = <0x40016800 0x400>;
+			interrupts = <135 0>, <136 0>;
+			interrupt-names = "ltdc", "ltdc_er";
+			clocks = <&rcc STM32_CLOCK(APB2, 26)>;
+			resets = <&rctl STM32_RESET(APB2, 26)>;
+			status = "disabled";
+		};
```
Since this seems like an addition, could it be split to a separate commit?

harristomy commented on dts/arm/st/u5/stm32u575Xg.dtsi:
```diff
@@ -0,0 +1,22 @@
+		/* SRAM1 + SRAM2 + SRAM3 */
+		reg = <0x20000000 DT_SIZE_K(768)>;
+	};
+
```
Added

harristomy commented on dts/arm/st/u5/stm32u5_usbotg_fs.dtsi:
```diff
@@ -0,0 +1,32 @@
+
+#include <st/u5/stm32u5.dtsi>
+
+#include <zephyr/dt-bindings/clock/stm32u5_clock.h>
```
Removed

harristomy reviewed (commented):
@mathieuchopstm comments addressed, cheers for the review

harristomy commented on dts/arm/st/u5/stm32u5.dtsi:
```diff
@@ -312,53 +312,53 @@
 			reg = <0x40004400 0x400>;
-			clocks = <&rcc STM32_CLOCK(APB1, 17U)>;
-			resets = <&rctl STM32_RESET(APB1L, 17U)>;
+			clocks = <&rcc STM32_CLOCK(APB1, 17)>;
```
moved.

harristomy commented on dts/arm/st/u5/stm32u5.dtsi:
```diff
@@ -218,71 +218,71 @@
 				#gpio-cells = <2>;
 				reg = <0x42021400 0x400>;
-				clocks = <&rcc STM32_CLOCK(AHB2, 5U)>;
+				clocks = <&rcc STM32_CLOCK(AHB2, 5)>;
```
moved.

harristomy commented on dts/arm/st/u5/stm32u5.dtsi:
```diff
@@ -218,71 +218,71 @@
 				#gpio-cells = <2>;
 				reg = <0x42022000 0x400>;
-				clocks = <&rcc STM32_CLOCK(AHB2, 8U)>;
+				clocks = <&rcc STM32_CLOCK(AHB2, 8)>;
```
moved.

harristomy commented on dts/arm/st/u5/stm32u5.dtsi:
```diff
@@ -753,35 +731,35 @@
 			compatible = "st,stm32-sdmmc";
 			reg = <0x420c8c00 0x400>;
-			clocks = <&rcc STM32_CLOCK(AHB2, 28U)>,
+			clocks = <&rcc STM32_CLOCK(AHB2, 28)>,
```
moved.

harristomy reviewed (commented):
Comments addressed

harristomy committed:
dts/kconfig: stm32u5: add f9 and clean up dts node locations

Adds skeleton dtsi for u5f9 for u5g9 to inherit from

Moves the peripheral nodes into dtsi's that actually has the peripheral
and includes them for SoC's higher in the series where applicable.

signed-off-by: Harris Tomy <harristomy@gmail.com>

harristomy committed:
dts: stm32u5: Removes trailing 'U's in dt props

Integers in devicetree are always signed.
See https://github.com/zephyrproject-rtos/zephyr/pull/89978#discussion_r2124113613

signed-off-by: Harris Tomy <harristomy@gmail.com>

etienne-lms commented on dts/arm/st/u5/stm32u535Xb.dtsi:
```diff
@@ -0,0 +1,28 @@
+	sram1: memory@28000000 {
+		/* SRAM4, low-power background autonomous mode */
+		reg = <0x28000000 DT_SIZE_K(16)>;
+	};
```
Sorry for this late comment.
These 2 nodes (the 3 SRAMs) are common to all stm32u53/54 SoCs.
Maybe they could be factorized in **stm32u535.dtsi**.

etienne-lms commented on dts/arm/st/u5/stm32u575Xg.dtsi:
```diff
@@ -0,0 +1,27 @@
+	sram1: memory@28000000 {
+		/* SRAM4, low-power background autonomous mode */
+		reg = <0x28000000 DT_SIZE_K(16)>;
+	};
```
Here also, these 2 nodes are common to stm32u575 and u585 SoCs.
Could be factorized in **stm32u575.dtsi** instead of being placed here and later replicated other stm32u575X* DTSI files.

etienne-lms commented on dts/arm/st/u5/stm32u599Xi.dtsi:
```diff
@@ -0,0 +1,26 @@
+		/* SRAM1 + SRAM2 + SRAM3 + SRAM5 */
+		reg = <0x20000000 DT_SIZE_K(2496)>;
+	};
+	sram1: memory@28000000 {
```
Could you add the missing empty line above?

etienne-lms commented on dts/arm/st/u5/stm32u5f9.dtsi:
```diff
@@ -0,0 +1,13 @@
+/ {
+	soc {
+		compatible = "st,stm32u5f9", "st,stm32u5", "simple-bus";
+	};
```
Missed that it would maybe be worth to consider also the SRAM6 (a diff between u59*/u5A* and u5F*/u5G*) in the `sram0` node?
Doing so would overwrite node sram0 from **stm32u599.dtsi**.

etienne-lms commented on dts/arm/st/u5/stm32u599.dtsi:
```diff
@@ -9,5 +9,24 @@
+			/* SRAM1 + SRAM2 + SRAM3 + SRAM5 */
+			reg = <0x20000000 DT_SIZE_K(2496)>;
+		};
+		sram1: memory@28000000 {
```
add an empty line above

etienne-lms reviewed (commented):
thanks for the updates

harristomy reviewed (commented):
🤞🏽

harristomy commented on dts/arm/st/u5/stm32u595.dtsi:
```diff
@@ -1,14 +1,23 @@
+		/* SRAM1 + SRAM2 + SRAM3 + SRAM5 */
+		reg = <0x20000000 DT_SIZE_K(2496)>;
+	};
+	sram1: memory@28000000 {
```
😭

harristomy commented on soc/st/stm32/stm32u5x/Kconfig.defconfig.stm32u5f9xx:
```diff
@@ -0,0 +1,11 @@
+# STMicroelectronics STM32U5GFXX MCU
```
🥲 sorry I tried using Github Codespaces for this PR but it feels like it often didn't save some changes (hence the number of previously resolved comments appearing again)

harristomy replied:
Out of curiosity, why are these files going to be deleted?
